core/src/test/scala/integration/kafka/api/RackAwareAutoTopicCreationTest.scala

```diff
@@ -35,31 +37,37 @@ class RackAwareAutoTopicCreationTest extends KafkaServerTestHarness with RackAwa
   overridingProps.put(KafkaConfig.NumPartitionsProp, numPartitions.toString)
   overridingProps.put(KafkaConfig.DefaultReplicationFactorProp, replicationFactor.toString)
 
+
   def generateConfigs =
     (0 until numServers) map { node =>
-      TestUtils.createBrokerConfig(node, zkConnect, enableControlledShutdown = false, rack = Some((node / 2).toString))
+      TestUtils.createBrokerConfig(node, zkConnectOrNull, enableControlledShutdown = false, rack = Some((node / 2).toString))
     } map (KafkaConfig.fromProps(_, overridingProps))
 
   private val topic = "topic"
 
-  @Test
-  def testAutoCreateTopic(): Unit = {
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumName)
+  @ValueSource(strings = Array("zk")) // TODO Partition leader is not evenly distributed in kraft mode, see KAFKA-15354
```

Comment:
Do we need to fix KAFKA-15354 first? Or can this ticket be resolved?

Reply:
I tried to fix it but we are using a different way to assign leader in kraft mode, the ideas behind the two algorithms are similar but the implementations are different. 
And the new one, `StripedReplicaPlacer`, is clearer, but it only ensure leader are distributed evenly across racks, and don't ensure leaders are evenly distributed across nodes. I wonder whether it is worth evolving it to be consistent with the older one. cc @cmccabe

Reply:
@mimaison I reverted this change for now since we are moving this to KAFKA-15354.